src/dune_rules/vimpl.ml

```diff
@@ -30,13 +30,7 @@ let make ~vlib ~impl ~vlib_modules ~vlib_foreign_objects =
     in
     Mode.cm_kind (if byte then Byte else Native)
   in
-  let vlib_obj_map =
-    Modules.obj_map vlib_modules ~f:(function
-      | Normal m -> m
-      | _ -> assert false)
-    |> Module.Obj_map.fold ~init:Module_name.Unique.Map.empty ~f:(fun m acc ->
-           Module_name.Unique.Map.add_exn acc (Module.obj_name m) m)
-  in
+  let vlib_obj_map = Modules.unique_map vlib_modules in
```

Comment:
⚠️ After this change, the assertion above will never be triggered anymore, as `Modules.unique_map` permits both `Imported_from_vlib` and `Impl_of_virtual_module` values. Should we add a special case for vlibs?

Reply:
I'm thinking, do we even need to compute this map right now? We're already saving `vlib_modules`, what about getting rid of this field?

Reply:
We don't need to, updated record in https://github.com/ocaml/dune/pull/7187/commits/955fab5f9bced5987da8265b16f4ca2bcafcfd03. My previous comment about the assertion going away still applies, I understand it's not a big deal.